src/h3lib/include/polygon.h

```diff
@@ -24,13 +24,61 @@
 #include "bbox.h"
 #include "geoCoord.h"
 #include "h3api.h"
+#include "linkedGeo.h"
+
+/** @struct IterableGeoLoop
+ *  @brief  Struct supporting polymorphic loop iteration
+ */
+typedef struct {
+    union {
+        const Geofence* geofence;            ///< optional Geofence
+        const LinkedGeoLoop* linkedGeoLoop;  ///< optional LinkedGeoLoop
+    };
+    int type;  ///< flag for type held by struct
+} IterableGeoLoop;
```

Comment:
This name really suggests an inheritance hierarchy. I am tempted to suggest the function pointer, cast to base class solution I saw in CPython. But I think an X-Macro is a better solution here.